frontend/src/components/RecordingDataGrid.tsx

```diff
@@ -1,52 +1,64 @@
-import {EuiBasicTable, EuiBasicTableColumn, EuiBasicTableProps} from '@elastic/eui';
+import {EuiBasicTable, EuiBasicTableColumn, EuiBasicTableProps, EuiButton, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiSearchBar, EuiSearchBarOnChangeArgs, EuiSpacer, RIGHT_ALIGNMENT} from '@elastic/eui';
 import {useLoaderData, useLocation, useNavigate} from 'react-router';
-import {Recording, useDeleteRecordingById, useFindAllRecordings} from '../api';
+import {Recording, useDeleteRecordingById, useFindAllRecordings, useFindImageById} from '../api';
 import prettyBytes from 'pretty-bytes';
-import ContextMenu from "./ContextMenu";
 import {useEffect, useState} from "react";
 import {useQueryClient} from "@tanstack/react-query";
 
-const tableColumns: EuiBasicTableColumn<Recording>[] = [
-  {
-    field: 'id',
-    name: 'Id',
-  },
-  {
-    field: 'name',
-    name: 'File Name',
-  },
-  {
-    field: 'recordingImage',
-    name: 'Image Name',
-  },
-  {
-    field: 'size',
-    name: 'Size',
-    render: (value: number) => prettyBytes(value, {maximumFractionDigits: 2}),
-  },
-  {
-    field: 'date',
-    name: 'Timestamp',
-  },
-]
-
 function RecordingDataGrid() {
   const navigate = useNavigate();
   const location = useLocation();
   const {isLoading, error, data} = useFindAllRecordings();
   const queryClient = useQueryClient();
   const deleteFunction = useDeleteRecordingById({mutation: {onSuccess: () => queryClient.invalidateQueries()}});
 
-  const deleteRecording = ({recordingId}: {recordingId: string}) => {
-    deleteFunction.mutate({recordingId: recordingId});
+  const deleteRecording = ({itemId}: {itemId: string}) => {
+    deleteFunction.mutate({recordingId: itemId});
   }
 
   useEffect(() => {
     if(location.state) {
-      deleteRecording({recordingId: location.state.recordingId});
+      deleteRecording({itemId: location.state.recordingId});
     }
   }, []);
 
+  function deleteActionPress (event: React.MouseEvent, item: Recording){
+    deleteRecording({itemId: item.id!})
+    event.stopPropagation();
+  }
+
+  const tableColumns: EuiBasicTableColumn<Recording>[] = [
+    {
+      field: 'id',
+      name: 'Id',
+    },
+    {
+      field: 'name',
+      name: 'File Name',
+    },
+    {
+      field: 'recordingImage',
+      name: 'Image Id',
+    },
+    {
+      field: 'size',
+      name: 'Size',
+      render: (value: number) => prettyBytes(value, {maximumFractionDigits: 2}),
+    },
+    {
+      align: RIGHT_ALIGNMENT,
```

Comment:
Recordings and Images both have this delete option, but Interactions don't. Is this intentional?